nx/lib/nx/defn.ex

```diff
@@ -686,6 +740,33 @@ defmodule Nx.Defn do
     end
   end
 
+  defp define_transform(kind, call, block, env) do
+    # Note name here is not necessarily an atom due to unquote(name) support
+    {name, args} = decompose_call!(kind, call, env)
+    arity = length(args)
+
+    defaults =
```

Comment:
we still need to handle defaults, guards and multiclause for deftransform. Are these in scope for this PR?

Reply:
Guards and multiclause yes, those should be easy. I would skip on defaults for now!

Reply:
Actually, let's do it all. It seems the logic for defaults is already ready in the compiler. :)